src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,38 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+interface Props {
+  onResetSelectedTodo: () => void;
+  selectedTodo: Todo;
+}
+
+export const TodoModal: React.FC<Props> = ({
+  onResetSelectedTodo,
+  selectedTodo,
+}) => {
+  const [loadingTodoModal, setLoadingTodoModal] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setLoadingTodoModal(true);
+    getUser(selectedTodo.userId)
+      .then(setUser)
+      .finally(() => setLoadingTodoModal(false));
+  }, [selectedTodo.userId]);
```

Comment:
You are initializing the state with props, which can lead to bugs and make the component harder to understand. It's better to initialize the state with a default value and then update it based on the props in the useEffect hook.